### PR TITLE
ci: fix buildkite analytics for e2e

### PR DESCRIPTION
**Pull request by jhchabran**
There was a bug in how the script was written for the E2E test, which prevented the API key to be properly assigned. 

There was also another bug that slipped in that messed the exit code handling code in the QA test suite. 

I have taken the opportunity of also improving the output, issuing a warning message when the test suite isn't working. 

It's probably a good candidate for sentry reporting as well, I'll add that in a lta

## Test plan

- CI Run and observing that: 
  - E2E test suite properly submits its report
  - QA test suite properly submits its report
  - Go test suite properly submits its report

### CODE REVIEW

File: dev/ci/integration/e2e/test.sh [hunk 1]
```diff
@@ -47,14 +47,22 @@ function integration_test() {
 EOF
   )
 
+  echo -e "\n--- :information_source: Uploading test results to Buildkite analytics"
+  set +e
   echo "$data" | curl \
+    --fail \
```
Comment (umpox) [inline dev/ci/integration/e2e/test.sh:53]: How come we have `--fail` here but not in the other scripts?

**Review by umpox — APPROVED** [review]

File: dev/ci/integration/e2e/test.sh (hunk 1 above)
Comment (jhchabran) [reply dev/ci/integration/e2e/test.sh:53]: Typical of me, fixing the other scripts. Thanks!

**Comment by bobheadxi** [comment]
> It's probably a good candidate for sentry reporting as well, I'll add that in a lta

Before you do this, I'd like to submit a proposal once you merge this PR that integrates test uploads with an export-scrape approach similar to how `bk.AnnotatedCmd` is done now! This way we don't have to update all these touch points and raw curls in scripts

**Review by bobheadxi — APPROVED** [review]
LGTM!

**Comment by jhchabran** [comment]
> > It's probably a good candidate for sentry reporting as well, I'll add that in a lta
> 
> Before you do this, I'd like to submit a proposal once you merge this PR that integrates test uploads with an export-scrape approach similar to how `bk.AnnotatedCmd` is done now! This way we don't have to update all these touch points and raw curls in scripts

Just FYI, Buildkite has planned to provide a `buildkite-agent analytics ...` thing, but that won't prevent us to actually the approach you're describing here. 

This also makes me think we should have a `ErrorReportedCmd` in the same way you added the `AnnotatedCmd`! (We may want to start having some options there rather than flat funcs, they're getting numerous :P)

**Pull request merged**